File: androidshared/src/main/java/org/odk/collect/androidshared/system/PermissionsChecker.java
package org.odk.collect.androidshared.system;

import android.content.Context;
import android.content.pm.PackageManager;

import androidx.core.content.ContextCompat;

public class PermissionsChecker {
    private final Context context;

    public PermissionsChecker(Context context) {
        this.context = context;
    }

    public boolean isPermissionGranted(String... permissions) {
        for (String permission : permissions) {
            if (ContextCompat.checkSelfPermission(context, permission) != PackageManager.PERMISSION_GRANTED) {
                return false;
            }
        }
        return true;
    }
}

File: permissions/src/main/java/org/odk/collect/permissions/PermissionsChecker.kt
package org.odk.collect.permissions

import android.content.Context
import android.content.pm.PackageManager
import androidx.core.content.ContextCompat

interface PermissionsChecker {
    fun isPermissionGranted(vararg permissions: String): Boolean
}

open class ContextCompatPermissionChecker(private val context: Context) : PermissionsChecker {

    override fun isPermissionGranted(vararg permissions: String): Boolean {
        for (permission in permissions) {
            if (ContextCompat.checkSelfPermission(context, permission) != PackageManager.PERMISSION_GRANTED) {
                return false
            }
        }
        return true
    }
}
